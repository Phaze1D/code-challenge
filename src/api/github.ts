import axios from 'axios';
import { SearchParams } from 'types';

const githubAxios = axios.create({
  baseURL: 'https://api.github.com',
  headers: {
    common: {
      Accept: 'application/vnd.github.v3+json'
    }
  }
});
export const __githubAxios = githubAxios;


export const setToken = (token: string) => {
  githubAxios.defaults.headers.common.Authorization = `token ${token}`;
};


export const searchRepos = (params: SearchParams) => {
  return githubAxios.get('/search/repositories', {params});
};
