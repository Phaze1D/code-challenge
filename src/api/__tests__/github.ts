import MockAdapter from 'axios-mock-adapter';
import { searchRepos, __githubAxios, setToken } from '../github';

describe('Github API Config Test', () => {
  const mock = new MockAdapter(__githubAxios);

  test('Should match correct config without auth', async () => {
    mock.onGet(/\S/).reply(200, {});
    const res = await searchRepos({q: 'reacts'});
    expect(res.config.headers).toMatchSnapshot();
    expect(res.config.baseURL).toMatchSnapshot();
  });

  test('Should match correct config with auth', async () => {
    setToken('falsetoken923908-4325');
    mock.onGet(/\S/).reply(200);
    const res = await searchRepos({q: 'reacts'});
    expect(res.config.headers).toMatchSnapshot();
    expect(res.config.baseURL).toMatchSnapshot();
  });
});
