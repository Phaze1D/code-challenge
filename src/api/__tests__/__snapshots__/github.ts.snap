// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github API Config Test Should match correct config with auth 1`] = `
Object {
  "Accept": "application/vnd.github.v3+json",
  "Authorization": "token falsetoken923908-4325",
}
`;

exports[`Github API Config Test Should match correct config with auth 2`] = `"https://api.github.com"`;

exports[`Github API Config Test Should match correct config without auth 1`] = `
Object {
  "Accept": "application/vnd.github.v3+json",
}
`;

exports[`Github API Config Test Should match correct config without auth 2`] = `"https://api.github.com"`;
